monte_carlo.py

```diff
@@ -43,7 +44,7 @@ def save(self, filename):
                 f.write(f"{state[0]},{state[1]},{state[2]},{state[3]},{action}\n")
     
 class MonteCarlo:
-
+    actions_memoization = {}
```

Comment:
maybe rename as 
valid_actions_memoization
for immediate understanding as I had to see changes in get_valid_actions() to see what it is used for